JoinColony/colonyDapp
# Coin Machine Buy Tokens Widget

## Description

This PR implements the "Buy Tokens" widget to the Coin Machine page, that allows the user the ability to purchase a certain amount of tokens in the current sale interval.

**Note** that this escalated is scope from when it was initially spec'd out, due to the previous team's failure to implement it's UI originally. See #2206 if you think you can find it.

**Testing**
- Create a colony
- Mint tokens _(1000+)_
- Unlock the token
- Install Coin Machine 
- Enable with _Default Values_
- Send 100+ tokens to the Coin Machine contract _(copy the contract address and create a payment action, with the address as a recipient)_
- Go buy some tokens on the Coin Machine page
- **Note that you can only buy a max of 100 tokens per period, so if you reach that limit, the buy widget will not allow you to buy more, until the next period starts _(and since we don't have UI for that yet, you won't be able to tell)_**

**Changes**
- [x] Cherry picked `QuestionMarkTooltip` from the #2413 branch
- [x] Added global `learnMore` message descriptor _(and replaced it where it was being used)_
- [x] Fixed `Chat` component shadow
- [x] Fixed core `Heading` `medium` appearance size letter spacing _(to be in line with the spec)_
- [x] Added `BuyTokens` component
- [x] Added Coin Machine sagas layout
- [x] Added Coin Machine `buyTokens` saga
- [x] Added Coin Machine redux actions and types
- [x] Added Coin Machine resolvers
- [x] Added Coin Machine GraphQL queries and typedefs

**Demo**
![demo-cm-buy-tokens](https://user-images.githubusercontent.com/1193222/127321413-4016ca8a-ef28-48a7-b323-b3d4bb33843e.gif)

Resolves DEV-245
